File: src/main/java/com/ject/studytrip/studylog/application/dto/StudyLogInfo.java
package com.ject.studytrip.studylog.application.dto;

import com.ject.studytrip.global.util.DateUtil;
import com.ject.studytrip.studylog.domain.model.StudyLog;

public record StudyLogInfo(
        Long studyLogId,
        String title,
        String content,
        String imageUrl,
        String createdAt,
        String updatedAt,
        String deletedAt) {
    public static StudyLogInfo from(StudyLog studyLog) {
        return new StudyLogInfo(
                studyLog.getId(),
                studyLog.getTitle(),
                studyLog.getContent(),
                studyLog.getImageUrl(),
                DateUtil.formatDateTime(studyLog.getCreatedAt()),
                DateUtil.formatDateTime(studyLog.getUpdatedAt()),
                DateUtil.formatDateTime(studyLog.getDeletedAt()));
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/application/dto/StudyLogInfo.kt
package com.ject.studytrip.studylog.application.dto

import com.ject.studytrip.global.util.DateUtil
import com.ject.studytrip.global.util.EntityExtensions.requireId
import com.ject.studytrip.studylog.domain.model.StudyLog

data class StudyLogInfo(
    val studyLogId: Long,
    val title: String,
    val content: String,
    val imageUrl: String?,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?,
) {
    companion object {
        fun from(studyLog: StudyLog): StudyLogInfo =
            StudyLogInfo(
                studyLog.id.requireId(),
                studyLog.title,
                studyLog.content,
                studyLog.imageUrl,
                DateUtil.formatDateTime(requireNotNull(studyLog.createdAt)),
                DateUtil.formatDateTime(requireNotNull(studyLog.createdAt)),
                studyLog.deletedAt?.let { DateUtil.formatDateTime(it) },
            )
    }
}
